### DISCUSSION TITLE
Can you issue commands/macros with a barcode scanner?

### DISCUSSION THREAD

**Discussion by Benjamin0001** [inventree/InvenTree#4456] [Q&A]
Hello, I'm new here to Inventree and GitHub.
I'm trying to find if a feature is available. I have seen in other programs that you can use a barcode scanner as part of the UI besides just entering ID numbers for stock or locations.
Like have a barcode that activates a set location to empty. Or one that opens the input field for a stock count.
It can be a very friendly UI feature to be able to jump immediately to the needed function just by scanning a "cheat sheet" of barcodes(or QR codes).
The barcode scanner would likely be peripheral to a PC or Android device.

Is there anything like that in Inventree? Is there a way to search for plugins or other solutions other people have made?

**Comment by SchrodingersGat** [comment]
@Benjamin0001 as far as I am aware no one has implemented a plugin to do this. But it should be very possible (and trivial) using the [Barcode Plugin Mixin](https://docs.inventree.org/en/latest/extend/plugins/barcode/)

Just write some code that does what you intend on receiving a particular barcode! If you do make something we can feature it on our [plugin page](https://inventree.org/plugins)

Let me know if you need any pointers developing this - it would be a great idea!

**Reply by Benjamin0001** [reply]
I appreciate the quick reply. I might try this. This might just be the thing to get me do some programing. I haven't done much so hopefully it won't be too hard. I only have coded maybe 100 lines for an Arduino thing I made using the Arduino web compiler.
Is VS Code a good place to start or is there something more noob friendly that can handle this?

**Reply by SchrodingersGat** [reply]
@Benjamin0001 I develop in VS-code, but just as a code editor.

As a starting point for plugin development, have  a look at how the [inventree-brother-plugin](https://github.com/inventree/inventree-brother-plugin) code is structured. You can use that project as a starting point!
